apps/web/src/pages/Skills/UpdateUserSkillPage.tsx

```diff
@@ -0,0 +1,657 @@
+  skill: Skill;
+  experiences: Experience[];
+  userSkill?: Maybe<UserSkill>;
+}
+
+const UpdateUserSkillForm = ({
+  userId,
+  experiences,
+  skill,
+  userSkill,
+}: UpdateUserSkillFormProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const navigate = useNavigate();
+  const [isFormDialogOpen, setFormDialogOpen] = React.useState<boolean>(false);
+  const [currentExperience, setCurrentExperience] =
+    React.useState<Experience | null>(null);
+  const skillName = getLocalizedName(skill.name, intl);
+  const skillDescription = getLocalizedName(skill.description, intl);
+  const hasUserSkill = notEmpty(userSkill);
+  const linkedExperiences = userSkill?.experiences?.filter(notEmpty);
+
+  const availableExperiences = experiences.filter(
+    (exp) =>
+      !linkedExperiences?.find(
+        (existingExperience) => existingExperience.id === exp.id,
+      ),
+  );
+
+  const isTechnical = skill.category === SkillCategory.Technical;
+  const levelGetter = isTechnical
+    ? getTechnicalSkillLevel
+    : getBehaviouralSkillLevel;
+  const levelDefinitionGetter = isTechnical
+    ? getTechnicalSkillLevelDefinition
+    : getBehaviouralSkillLevelDefinition;
+  const levelOptions = getSortedSkillLevels().map((skillLevel) => ({
+    value: skillLevel,
+    label: <strong>{intl.formatMessage(levelGetter(skillLevel))}</strong>,
+    contentBelow: (
+      <p data-h2-margin="base(x.15, 0, x.5, x1)">
+        {intl.formatMessage(levelDefinitionGetter(skillLevel))}
+      </p>
+    ),
+  }));
+
+  const [{ fetching: creating }, executeCreateMutation] =
+    useCreateUserSkillMutation();
+  const [{ fetching: updating }, executeUpdateMutation] =
+    useUpdateUserSkillMutation();
+  const [{ fetching: deleting }, executeDeleteMutation] =
+    useDeleteUserSkillMutation();
+  const mutating = creating || updating || deleting;
+
+  const handleSuccess = (msg?: React.ReactNode) => {
+    toast.success(
+      msg ||
+        intl.formatMessage({
+          defaultMessage: "Successfully updated skill!",
+          id: "Vfa3Ek",
+          description: "Message displayed when a user updates a skill",
+        }),
+    );
+    navigate(paths.skillLibrary());
+  };
+
+  const handleError = (msg?: React.ReactNode) => {
+    toast.error(
+      msg ||
+        intl.formatMessage({
+          defaultMessage: "Error: updating skill failed",
+          id: "CUxHd8",
+          description:
+            "Message displayed to user after skill fails to be updated.",
+        }),
+    );
+  };
+
+  const handleSubmit = (values: FormValues) => {
+    if (hasUserSkill) {
+      executeUpdateMutation({
+        id: userSkill.id,
+        userSkill: values,
+      })
+        .then(() => handleSuccess())
+        .catch(() => handleError());
+    } else {
+      executeCreateMutation({
+        userId,
+        skillId: skill.id,
+        userSkill: values,
+      })
+        .then(() => handleSuccess())
+        .catch(() => handleError());
```

Comment:
Yeah, that is weird but we technically do allow this 😅 Though, you raise a good point.

@tristan-orourke Should we allow users to link an experience without first claiming the skill and setting their level and if it is current or not?

Reply:
Yes, this is intended. If a user links an experience to a skill, it gets added to their SkillLibrary automatically without a level or other metadata. If they go to the SkillLibrary page, they'll see `N/A` for those fields in the table.

Reply:
Just a heads up... It's an edge case, but the user will be able to connect an experience, then remove that experience, then leave to the skill library page and it could have a userSkill with "0 experiences" and "NA" skill level.